logging/bin/deploy_kibana_opensearch.sh

```diff
@@ -0,0 +1,136 @@
+
+if [ "$OPENSEARCHDASH_ENABLE" != "true" ]; then
+  log_verbose "Environment variable [OPENSEARCHDASH_ENABLE] is not set to 'true'; exiting WITHOUT deploying OpenSearch Dashboards"
+  exit 0
+fi
+
+set -e
+
+#
+# check for pre-reqs
+#
+
+# Confirm namespace exists
+if [ "$(kubectl get ns $LOG_NS -o name 2>/dev/null)" == "" ]; then
+  log_error "Namespace [$LOG_NS] does NOT exist."
+  exit 1
+fi
+
+# get credentials
+export ES_KIBANASERVER_PASSWD=${ES_KIBANASERVER_PASSWD}
+
+# Create secrets containing internal user credentials
+create_user_secret internal-user-kibanaserver kibanaserver "$ES_KIBANASERVER_PASSWD"  managed-by=v4m-es-script
+
+# Verify cert-manager is available (if necessary)
+if verify_cert_manager $LOG_NS kibana; then
+  log_debug "cert-manager check OK"
+else
+  log_error "One or more required TLS certs do not exist and cert-manager is not available to create the missing certs"
+  exit 1
+fi
+
+# Create/Get necessary TLS certs
+apps=( kibana )
+create_tls_certs $LOG_NS logging ${apps[@]}
+
+
+# Need to retrieve these from secrets in case secrets pre-existed
+export ES_ADMIN_USER=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.username}" |base64 --decode)
+export ES_ADMIN_PASSWD=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.password}" |base64 --decode)
+
+
+# enable debug on Helm via env var
+export HELM_DEBUG="${HELM_DEBUG:-false}"
+
+if [ "$HELM_DEBUG" == "true" ]; then
+  helmDebug="--debug"
+fi
+
+KB_KNOWN_NODEPORT_ENABLE=${KB_KNOWN_NODEPORT_ENABLE:-false}
+
+if [ "$KB_KNOWN_NODEPORT_ENABLE" == "true" ]; then
+   KIBANA_PORT=31034
+   log_verbose "Setting Kibana service NodePort to $KIBANA_PORT"
+   nodeport_yaml=logging/es/opensearch/osd_helm_values_nodeport_opensearch.yaml
+else
+   nodeport_yaml=$TMP_DIR/empty.yaml
+   log_debug "Kibana service NodePort NOT changed to 'known' port because KB_KNOWN_NODEPORT_ENABLE set to [$KB_KNOWN_NODEPORT_ENABLE]."
+fi
+
+
+# OpenSearch Dashboards user customizations
+OSD_USER_YAML="${OSD_USER_YAML:-$USER_DIR/logging/user-values-osd-opensearch.yaml}"
+if [ ! -f "$OSD_USER_YAML" ]; then
+  log_debug "[$OSD_USER_YAML] not found. Using $TMP_DIR/empty.yaml"
+  OSD_USER_YAML=$TMP_DIR/empty.yaml
+fi
+
+# Require TLS into OpenSearch Dashboards (nee Kibana)?
+LOG_KB_TLS_ENABLE=${LOG_KB_TLS_ENABLE:-false}
+
+# Enable TLS for East/West Kibana traffic (inc. requiring HTTPS from browser if using NodePorts)
+if [ "$LOG_KB_TLS_ENABLE" == "true" ]; then
+   # w/TLS: use HTTPS in curl commands
+   KB_CURL_PROTOCOL=https
+   log_debug "TLS enabled for Kibana"
+else
+   # w/o TLS: use HTTP in curl commands
+   KB_CURL_PROTOCOL=http
+   log_debug "TLS not enabled for Kibana"
+fi
+#(Re)Create secret containing OSD TLS Setting
+kubectl -n $LOG_NS delete secret          v4m-osd-tls-enabled  --ignore-not-found
+kubectl -n $LOG_NS create secret generic  v4m-osd-tls-enabled  --from-literal enable_tls="$LOG_KB_TLS_ENABLE"
+
```

Comment:
Should we use an annotation like we did in ODFE?

Reply:
After reviewing the code, I realized the secret is needed elsewhere (it sets an env var inside the OSD pod indicating whether TLS is enabled or not) and is already in place...so we're using it here rather than adding an annotation.